azureclient/credentialprovider/cli_flags.go

```diff
@@ -0,0 +1,113 @@
+package credentialprovider
+
+import (
+	"context"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CLIFlagsCredentialProviderConfig struct {
+	CtrlClient ctrl.Client
+	Logger     micrologger.Logger
+
+	ManagementClusterClientID       string
```

Comment:
I am not sure if, instead of passing values here, I should pass the `flag` object and do the parsing of cli arguments inside this module. WDYT?